CMakeLists.txt

```diff
@@ -207,6 +208,20 @@ else()
   message(STATUS "Parallelproj projector support disabled or not available.")
 endif()
 
+if(NOT DISABLE_STIR_CUDA)
+  find_package(CUDAToolkit)
+  if (CUDAToolkit_FOUND)
+    set(CMAKE_CUDA_ARCHITECTURES "all")
+    find_package(CUDA REQUIRED)
+    include_directories("${CUDA_INCLUDE_DIRS}")
+    set(STIR_WITH_STIR_CUDA ON)
+    if(STIR_WITH_STIR_CUDA)
+      enable_language(CUDA)
+      message(STATUS "STIR CUDA support enabled. Using CUDA version ${CUDAToolkit_VERSION}.")
+    endif()
+  endif()
```

Comment:
I think this is the other way around. I'd probably use https://cmake.org/cmake/help/latest/module/CheckLanguage.html#module:CheckLanguage first, if present, then `enable_language`, then [`find_package(CUDAToolkit)`](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html) which probably needs a `REQUIRED`.

Before all this, I'd do a check on CMAKE_VERSION. Certainly at least 3.18, but 3.23 if we want to use `"all" for 
[CMAKE_CUDA_ARCHITECTURES](https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html#prop_tgt:CUDA_ARCHITECTURES) (which we do). If the CMake version is too old, generate FATAL_ERROR with the message to set `DISABLE_STIR_CUDA=ON`.

Reply:
Also, don't use `include_directories` but `target_include_directories`

Reply:
done here: https://github.com/UCL/STIR/pull/1408/commits/d3f37324f65127673b04bfbaad1952c470edfab5

Reply:
It's this line:
```
if (STIR_WITH_CUDA)
  target_include_directories(stir_registries PRIVATE ${CUDA_INCLUDE_DIRS})
endif()
```